third-party/gmp/Makefile

```diff
@@ -28,7 +28,7 @@ CHPL_GMP_CFG_OPTIONS += --enable-static --disable-shared
 endif
 
 CHPL_GMP_ABI_ARG =
-ifeq (linux32, $(CHPL_MAKE_TARGET_PLATFORM))
+ifeq ($(CHPL_MAKE_TARGET_PLATFORM), $(filter $(CHPL_MAKE_TARGET_PLATFORM),linux32 cygwin32))
```

Comment:
I'm not opposed to this change to get cygwin32 working, but it is there a more general way to do this to avoid having to this again for things like bsd32 (if we actually make a platform like that.)

For instance could we do something use platform.architecture() to determine  32 vs 64 bit instead of platform name. 

I checked on linux64, linux32, cygwin64, cygwin32, and my mac, and it seems like it should work.